packages/cmd/src/CommandProcessor.ts

```diff
@@ -370,6 +365,15 @@ export class CommandProcessor {
 
         // Build the response object, base args object, and the entire array of options for this command
         // Assume that the command succeed, it will be marked otherwise under the appropriate failure conditions
+        if (params.arguments.dcd) {
+            // NOTE(Kelosky): we adjust `cwd` and do not restore it, so that multiple simultaneous requests from the same
+            // directory will operate without unexpected chdir taking place.  Multiple simultaneous requests from different
+            // directories may cause unpredictable results
+            process.chdir(params.arguments.dcd as string)
+
+            // reinit config for daemon client directory
+            ImperativeConfig.instance.config = await Config.load(ImperativeConfig.instance.rootCommandName, ImperativeConfig.instance.config.opts);
```

Comment:
This makes sense for now. 🙂 I'm wondering if in the future we'll want a more efficient way to alter the project directory of the Config class - assuming that global directory never changes.

Reply:
> a more efficient way to alter the project directory of the Config class

Having this would also be helpful for fixing #506

Reply:
im still thinking about this - but the i think there is a flaw in this logic somewhere ... i'll keep testing but i might need to fix a bug here